backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/internal/BpkRatingComponents.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import net.skyscanner.backpack.compose.rating.BpkRatingScale
+import net.skyscanner.backpack.compose.rating.BpkRatingSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@Composable
+internal fun BpkRatingNumbers(
+  value: Float,
+  scale: BpkRatingScale?,
+  size: BpkRatingSize,
+  modifier: Modifier = Modifier,
+) {
+  Row(modifier = modifier) {
+
+    val numberFormat = remember(LocalConfiguration.current.locales) { DecimalFormat("#0.0") }
+
+    BpkText(
+      modifier = Modifier.alignByBaseline(),
+      text = formatValue(value, scale, numberFormat),
+      style = when (size) {
+        BpkRatingSize.Base -> BpkTheme.typography.label1
+        BpkRatingSize.Large -> BpkTheme.typography.hero5
+      },
+      color = BpkTheme.colors.textPrimary,
+      maxLines = 1,
+      overflow = TextOverflow.Ellipsis,
+    )
+
+    if (scale != null) {
+      BpkText(
+        modifier = Modifier.alignByBaseline(),
+        text = "/${scale.range.endInclusive.toInt()}",
+        style = when (size) {
+          BpkRatingSize.Base -> BpkTheme.typography.caption
+          BpkRatingSize.Large -> BpkTheme.typography.bodyDefault
+        },
+        color = BpkTheme.colors.textSecondary,
+        maxLines = 1,
+        overflow = TextOverflow.Ellipsis,
+      )
+    }
+
+  }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+internal fun BpkRatingTitle(
+  modifier: Modifier = Modifier,
+  content: @Composable () -> Unit,
+) {
+  CompositionLocalProvider(
+    LocalTextStyle provides BpkTheme.typography.heading5,
+    LocalContentColor provides BpkTheme.colors.textPrimary,
+    LocalContentAlpha provides 1f,
+  ) {
+    Box(
+      modifier = modifier.heightIn(max = BpkSpacing.Lg),
+      contentAlignment = Alignment.CenterStart,
+    ) {
+      // a little trick to provide baseline params for the custom layouts with invisible text
+      BpkText(
+        text = "",
+        modifier = modifier
+            .alpha(0f)
+            .semantics { invisibleToUser() },
+      )
+      content()
+    }
+  }
+}
+
+@Composable
+internal fun BpkRatingSubtitle(
+  subtitle: String,
+  modifier: Modifier = Modifier,
+) {
+  BpkText(
+    modifier = modifier,
+    text = subtitle,
+    style = BpkTheme.typography.bodyDefault,
+    color = BpkTheme.colors.textSecondary,
+    maxLines = 1,
+    overflow = TextOverflow.Ellipsis,
+  )
+}
+
+private fun formatValue(value: Float, scale: BpkRatingScale?, format: DecimalFormat): String {
+  val coerced = if (scale != null) value.coerceIn(scale.range) else value
+  return format.format(coerced)
```

Comment:
how is this rounding? i remember the existing component had a specific rounding we had to use

Reply:
https://github.com/Skyscanner/backpack-android/blob/936cfcb629c53317c407bcd626581bdab6c80c38/Backpack/src/main/java/net/skyscanner/backpack/rating/internal/RatingScore.kt#L38

This is what we use for rounding. Added it to the code